recipe/meta.yaml

```diff
@@ -1,42 +1,76 @@
 {% set name = "pytorch-forecasting" %}
-{% set version = "0.10.2" %}
+{% set version = "0.10.3" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/p/pytorch_forecasting/pytorch_forecasting-{{ version }}.tar.gz
-  sha256: 069af34c17eea2edf5c11c79931097427b88cef8a862457818a03464260e5534
+  url: https://github.com/jdb78/pytorch-forecasting/archive/refs/tags/v0.10.3.tar.gz
+  sha256: 026a0498cd2444e38bb7db4c998d1dda5ee4c5e6418cce9d825eae06f892a1c6
+  patches:
+    - fix_versions.patch
 
 build:
-  noarch: python
   number: 0
-  script: {{ PYTHON }} -m pip install . -vv
+  # pyarrow and fastparquet as test dependencies are not available on s390x
+  skip: True  # [py<38 or py>310 or s390x]
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
 
 requirements:
+  build:
+    - m2-patch  # [win]
+    - patch     # [not win]
   host:
+    # llvm-openmp 14.0.6 added here only for intel-openmp <20.23.1 compatibility on osx-64. 
+    # Check if the issue persists with new releases
+    - llvm-openmp 14.0.6  # [osx and x86_64]
     - pip
-    - python >=3.6
-    - poetry >=1.0.5
+    - poetry-core 1.5.1
+    - poetry-dynamic-versioning 0.23.0
+    - python
   run:
-    - pytorch-lightning >=1.0.4
-    - optuna >=2.3.0
-    - python >=3.6.1
-    - pandas >=1.1.0
-    - scikit-learn >=0.23.0
-    - scipy >=1.5.2
-    - cloudpickle >=1.5.0
-    - statsmodels >=0.11.0
-    - pytorch >=1.4
     - matplotlib-base >=3.3.1
-    - prettytable <0.8
+    - optuna >=2.3.0
+    - pandas >=1.3.0
+    - python
+    - pytorch >=1.7,<2.0
+    # In pytorch-lightning v2.0.0 module named 'pytorch_lightning.core.lightning' was removed
+    - pytorch-lightning >=1.2.4,<2.0.0
+    - scikit-learn >=0.24.0,<1.2
+    - scipy >=1.8
+    - statsmodels
+  run_constrained:
+    # intel-openmp 2023.1 and llvm-openmp 14.0.6 appear not be compatible on osx-64
+    # leading to segfault in a lot of tests.
+    # Check if the issue persists with new releases
+    - intel-openmp <2023.1  # [osx and x86_64]
 
 test:
   imports:
     - pytorch_forecasting
+  source_files:
+    - tests
   requires:
+    - fastparquet
+    - networkx
+    # Use numpy <1.24 to avoid error numpy hasn't np.float 
+    - numpy <1.24
```

Comment:
Possibly add here too:

blas=*=openblas # [osx and x86_64]

to pick openblas (and hence no intel-openmp) variant of numpy.

Reply:
Added